apps/astarte_realm_management/lib/astarte_realm_management/queries.ex

```diff
@@ -558,21 +558,15 @@ defmodule Astarte.RealmManagement.Queries do
     Repo.fetch_all(query, prefix: keyspace, consistency: :quorum)
   end
 
-  def delete_devices_with_data_on_interface(client, interface_name) do
-    devices_statement = "DELETE FROM kv_store WHERE group=:group_name"
+  def delete_devices_with_data_on_interface(realm_name, interface_name) do
+    keyspace = Realm.keyspace_name(realm_name)
+    group_name = "devices-with-data-on-interface-#{interface_name}-v0"
 
-    devices_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(devices_statement)
-      |> DatabaseQuery.put(:group_name, "devices-with-data-on-interface-#{interface_name}-v0")
-      |> DatabaseQuery.consistency(:each_quorum)
+    query = from KvStore, where: [group: ^group_name]
 
-    with {:ok, _result} <- DatabaseQuery.call(client, devices_query) do
-      :ok
-    else
-      {:error, reason} ->
-        _ = Logger.warning("Database error: #{inspect(reason)}.", tag: "db_error")
-        {:error, :database_error}
+    case Repo.delete_all(query, prefix: keyspace, consistency: :each_quorum) do
+      {0, _} -> {:error, :not_found}
+      {_n, _} -> :ok
```

Comment:
My previous point was: are we sure that there is a match in the key value store?
If it's ok that no match exist, this code is returning an error even though interface deletion was successful.

Reply:
Ah you're right, re-looking at previous code it did not check for a match, in the last commit I solved this